### ISSUE TITLE
HkTimerKick runs, but it doesn't trigger the Timer function in each SpaceObject

### ISSUE THREAD

**Issue by ConradWeiser** [LazDisco/FLHook#12]
```
void HkTimerCheckKick()
{
	for (auto& spaceObject : spaceObjects)
	{
		spaceObject.second->Timer(timeInMS());
	}
}
```

Relevant code above. While I can verify that the method this loop is inside of runs every second, for some reason, despite compiling, the timer function does not run.

Is the spaceObject list empty? It shouldn't be?

**Comment by ConradWeiser** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [HkTimerKick runs, but it doesn't trigger the Timer function in each SpaceObject](https://trello.com/c/vjpyKeUr/24-hktimerkick-runs-but-it-doesnt-trigger-the-timer-function-in-each-spaceobject)

**Comment by ConradWeiser** [comment]
Resolved in b13c7ca3f16283934c51a1be5a2aba81e4bf2695
spaceObjects map operations were acting funky while being done inside of the SpaceObject class - A method in Main.cpp was created to handle it called `AddSpaceObjectToList(SpaceObject*)` - Please use it.
